lib/eecircuit-engine/run-simulation.ts

```diff
@@ -0,0 +1,63 @@
+import type * as EecircuitEngine from "lib/types/eecircuit-engine"
+import { promises as fs, existsSync } from "node:fs"
+import path from "node:path"
+import os from "node:os"
+
+let sim: EecircuitEngine.Simulation | null = null
+
+const fetchSimulation = async (): Promise<
+  typeof EecircuitEngine.Simulation
+> => {
+  // Using a predictable path in tmp to avoid re-downloading.
+  // This is a workaround for Bun's test environment not handling URL imports.
+  const tempFilePath = path.join(os.tmpdir(), "eecircuit-engine-1.5.2.mjs")
+
+  if (!existsSync(tempFilePath)) {
+    const url = "https://cdn.jsdelivr.net/npm/eecircuit-engine@1.5.2/+esm"
```

Comment:
I think we can bundle this inside the CLI since the CLI isn't used in browser- i don't mind this approach though, and actually if we save it to e.g. `${USER_CONFIG_DIR}/.tscircuit/eecircuit-engine-build.js` or something it could be nice to dynamically load some of these dependencies, as long as we don't dynamically load them every time

Don't worry about this too much, i think we can adjust this in the future. It's a nice implementation, just has some loading cost